pkg/services/ngalert/state/state_test.go

```diff
@@ -71,25 +71,14 @@ func TestNeedsSending(t *testing.T) {
 			},
 		},
 		{
-			name:        "state: normal + resolved sends after a minute",
+			name:        "state: normal + resolved should send without waiting",
 			resendDelay: 1 * time.Minute,
 			expected:    true,
 			testState: &State{
 				State:              eval.Normal,
 				Resolved:           true,
 				LastEvaluationTime: evaluationTime,
-				LastSentAt:         evaluationTime.Add(-1 * time.Minute),
-			},
-		},
-		{
-			name:        "state: normal + resolved does _not_ send after 30 seconds (before one minute)",
```

Comment:
Before this change `eval.Normal` states that were resolved would have to wait the interval provided to `NeedsSending`. However, because a resolved state has `Resolved` set as true for the first evaluation, and false for all subsequent evaluations, this meant that resolved states were never sent to the Alertmanager. Instead, what I think we should do is send the resolved state to the Alertmanager as soon as it is resolved, rather than attempt to wait the interval.

Reply:
What's the downside of waiting for the interval? Does it make the code too complicated?

Reply:
Should we instead have the `Resolved` value persist over time?

Reply:
If we want to wait for the interval we need to make `Resolved` persist over time. However, since state **N** does not know about state **N-1**, we don't know if this `Resolved` resolved is the first resolved state, or the **N+1**<sup>th</sup> resolved state.